spec/features/audit/report/csv_export_spec.rb

```diff
@@ -13,99 +7,140 @@ def content_disposition
+    expect(@audit_report_page).to be_displayed
+    expect(@audit_report_page).to have_no_content('2 Content items')
+  end
 
-      click_on "Apply filters"
-      expect(page).to have_content("Example 1")
-      expect(page).to have_no_content("Example 2")
+  def and_i_see_the_one_content_item_assigned_to_me
+    expect(@audit_report_page).to have_content('1 Content items')
+  end
 
-      click_link "Audit progress"
+  def when_i_export_an_audit_report
+    @audit_report = ContentAuditTool.new.audit_report_page
+    @audit_report.load(content_id: Content::Item.last.content_id)
+    @audit_report.export_to_csv.click
+  end
 
-      click_link "Export filtered audit to CSV"
-      expect(content_disposition).to include(
-        'filename="Transformation_audit_report_CSV_download.csv"',
-      )
-      expect(page).to have_content("Example 1")
-      expect(page).to have_no_content("Example 2")
-    end
+  def then_i_receive_the_report_in_csv_format
+    expect(content_type).to eq("text/csv")
+    expect(content_disposition).to start_with("attachment")
+    expect(content_disposition).to include(
+      'filename="Transformation_audit_report_CSV_download.csv"',
+    )
+    expect(@audit_report).to have_content("Title,URL")
+    expect(@audit_report).to have_content("Example 1,https://gov.uk/example1")
+  end
+
+  def and_filter_is_applied_to_show_only_hmrc_content_items
+    @audit_report = ContentAuditTool.new.audit_report_page
+    @audit_report.load(content_id: Content::Item.last.content_id)
+    @audit_report.organisations.select 'HMRC'
+    @audit_report.apply_filters.click
   end
 
-  context "Multiple pages of content items are in the database" do
-    let!(:content_items) { create_list(:content_item, 110) }
+  def when_i_export_the_reports
+    @audit_report.export_to_csv.click
+  end
 
-    scenario "Exporting an unfiltered audit to CSV with all the content items" do
-      visit audits_report_path
-      select "Anyone", from: "allocated_to"
-      click_on "Apply filters"
-      click_link "Export filtered audit to CSV"
+  def and_there_are_multiple_pages_of_unfiltered_content_items
+    create_list(:content_item, 100)
+  end
 
-      csv = CSV.parse(page.body, headers: true)
+  def when_i_export_all_the_reports
+    visit audits_report_path
+    select "Anyone", from: "allocated_to"
+    click_on "Apply filters"
+    click_link "Export filtered audit to CSV"
+  end
 
-      number_of_metadata_rows = 1
-      expect(csv.count).to eq(content_items.count + number_of_metadata_rows)
+  def then_i_receive_an_audit_progress_report_for_all_content_items_in_csv_format
+    csv = CSV.parse(page.body)
+    number_of_metadata_rows = 1
+    expect(csv.count).to eq(Content::Item.count + number_of_metadata_rows)
+  end
+
+  def when_i_filter_the_content_page_for_content_items_with_no_one_assigned_to_them
+    @audit_content_page = ContentAuditTool.new.audit_content_page
+    @audit_report_page = ContentAuditTool.new.audit_report_page
```

Comment:
Why do we need `@audit_report_page` here?

Reply:
It is needed because we navigate to the audit report page in this test, however I had this line in the wrong method 😔 Have moved it to the `when_i_navigate_to_audits_report_page` method.